.travis.yml

```diff
@@ -1,21 +1,40 @@
 ---
+sudo: required
+dist: trusty
+
 language: python
 python: "2.7"
 
+services:
+  - docker
+
 cache: pip
 
+addons:
+  apt:
+    packages:
+      - docker-engine
+      - sshpass
+      - software-properties-common
+      - python-software-properties
+
+
 install:
-  - pip install ansible ansible-lint yamllint
+  - pip install ansible ansible-lint docker-py yamllint
+  - ansible-galaxy install -r test/requirements.yml -p roles/
 
 before_script:
-  - find . -name '*.yml' -print0 | xargs -n1 -0 yamllint -c .yamllint.yml
+  - find . -path ./roles/provision_docker -prune -o  -name '*.yml' -print0 | xargs -n1 -0 yamllint -c .yamllint.yml
   - ansible-lint site.yml
-
-script:
   - ansible-playbook --syntax-check -i inventories/localhost site.yml
   - ansible-playbook --list-hosts -i inventories/localhost site.yml
   - ansible-playbook --list-tags -i inventories/localhost site.yml
   - ansible-playbook --list-tasks -i inventories/localhost site.yml
 
+script:
+  - ansible all -i "127.0.0.1," -m lineinfile -a "regexp=^#host_key_checking dest=/etc/ansible/ansible.cfg line='host_key_checking = False'" -c local
+  - ansible-playbook -i inventories/localhost test/main.yml --syntax-check
```

Comment:
The localhost inventory is being used to connect to the localhost to provision the Docker container. After that, it is using a dynamic inventory to connect to the container.